ext/TrixiMakieExt.jl

```diff
@@ -0,0 +1,18 @@
+# Package extension for adding Makie-based features to Trixi.jl
+module TrixiMakieExt
+
+# Required for visualization code
+using Makie: Makie, GeometryBasics
+
+# Use all exported symbols to avoid having to rewrite `recipes_makie.jl`
+using Trixi
+
+# Use additional symbols that are not exported
+using Trixi: PlotData2DTriangulated, TrixiODESolution, PlotDataSeries, ScalarData, @muladd
+
+# Import functions such that they can be extended with new methods
+import Trixi: iplot, iplot!
+
+include("../src/visualization/recipes_makie.jl")
```

Comment:
Why did you not move the corresponding plotting code to this file?

Reply:
At the moment, this PR leaves a lot "unfinished" and in between a proper transition to Pkg Extensions and the status quo. I therefore wanted to keep this PR minimally invasive. We can, however, also go all in and prepare everything for when we have Julia v1.9 as the minimum version.

I'm fine either way.

Reply:
We can move all Makie code to this file and still keep backwards compatibility. The required changes would be

1. Change
```julia
# Required for visualization code
using Makie: Makie, GeometryBasics
```
in this file to
```julia
# Required for visualization code
# We do not check `isdefined(Base, :get_extension)` since Julia v1.9.0
# does not load package extensions when their dependency is loaded from
# the main environment
if VERSION >= v"1.9.1"
    using Makie: Makie, GeometryBasics
else
    using ..Makie: Makie, GeometryBasics
end
```

2. Create empty functions `iplot` and `iplot!` in Trixi.jl unconditionally
3. Move the code from `src/visualization/recipes_makie.jl` to the place where we have `include("../src/visualization/recipes_makie.jl")` now
4. Change the Requires.jl block from
```julia
  @static if !(VERSION >= v"1.9.1")
    @require Makie="ee78f7c6-11fb-53f2-987a-cfe4a2b5a57a" begin
      include("visualization/recipes_makie.jl")
      using .Makie: Makie, GeometryBasics
      export iplot, iplot! # interactive plot
    end
```
to
```julia
  @static if !(VERSION >= v"1.9.1")
    @require Makie="ee78f7c6-11fb-53f2-987a-cfe4a2b5a57a" begin
      include("../ext/TrixiMakieExt.jl")
    end
```

This would keep all Makie code in a single file. I don't have a strong opinion on this, but it could be easier to understand.

Reply:
Done in a40327f0. Let's see what CI says.